portfolio/src/main/webapp/calendar.js

```diff
@@ -69,14 +103,70 @@ function getTimeObject(timeString) {
   return new Date(currentYear, currentMonth, currentDate, timeHour, timeMinute);
 }
 
-/** Creates a single element for a new calendar event. */
+/** Creates a card element for a new calendar event. */
+/** 
+ * TODO(hollyyuqizheng) Look into refactoring the card creation code 
+ * into a JavaScript class. This card element will be used in different 
+ * sections on the UI, so this refactoring will make things simpler. 
+ */
 function updateCalendarEventList(newCalendarEvent) {
-  const newEventElement = document.createElement('li');
-  newEventElement.innerHTML = 'New Event: ' + newCalendarEvent.name +
-      '<br>starts at ' + newCalendarEvent.startTime + '<br>ends at ' +
-      newCalendarEvent.endTime;
+  const newEventCard = document.createElement('div');
+  newEventCard.classList.add('card');
+
+  const cardBody = document.createElement('div');
+  cardBody.classList.add('card-body');
+  newEventCard.appendChild(cardBody);
+
+  const cardTitle = document.createElement('h4');
+  cardTitle.classList.add('card-title');
+  cardTitle.innerText = newCalendarEvent.name;
+  cardBody.appendChild(cardTitle);
+
+  const startTimeText = document.createElement('p');
+  startTimeText.classList.add('card-text');
+  startTimeText.innerText = newCalendarEvent.startTime;
+  cardBody.appendChild(startTimeText);
+
+  const endTimeText = document.createElement('p');
+  endTimeText.classList.add('card-text');
+  endTimeText.innerText = newCalendarEvent.endTime;
+  cardBody.appendChild(endTimeText);
+
+  const deleteButton = document.createElement('button');
+  deleteButton.classList.add('btn');
+  deleteButton.classList.add('btn-danger');
+  deleteButton.innerText = 'Delete this event';
+  cardBody.appendChild(deleteButton);
 
   const eventList = document.getElementById('new-event-list');
   eventList.innterHTML = '';
-  eventList.appendChild(newEventElement);
+  eventList.appendChild(newEventCard);
+
+  // The delete button removes the event's card from the UI.
+  deleteButton.onclick = function(newEventCard) {
+    newEventCard.target.closest('div.card').remove();
+  }
+}
+
+/** Collects and returns all the events currently displayed on the UI. */
+function collectAllEvents() {
+  // A set for all calendar events displayed on the UI.
+  // Each element in this set is a Json string.
+  var allEvents = [];
+
+  const eventList = document.getElementById('new-event-list');
+
+  // Looks at each event card and scrapes the event's name and
+  // start and end times from the HTML elements.
+  // Add all event information to a set of all Json strings.
+  eventList.childNodes.forEach((eventCard) => {
+    const eventCardBody = eventCard.childNodes[0];
+    const eventName = eventCardBody.childNodes[0].innerText;
+    const startTime = new Date(eventCardBody.childNodes[1].innerText);
```

Comment:
Consider using jquery for date picker.
https://jqueryui.com/datepicker/

But feel free to leave a TODO for the date picker.

Reply:
I see the benefit of using date picker on the UI, but for this function specifically, I was more hoping to extract relevant information from each card element on the page, so using date picker seems to not help in this regard here.